src/models/FieldTypes.js

```diff
@@ -0,0 +1,17 @@
+const FieldTypes = Object.freeze([
```

Comment:
We should use a ENUM notation for the variables : `BACK_END` instead of `Back-End`, either here or the DB, assuming this will be there in the future, so that we have the freedom to work with translations without much changes to the code, as the variables will be the same, and only the display changes depending on language

Reply:
I understand the idea, but am not really sure if this is all that relevant: it's a string all the same and the front-end will have to parse things back and forth if this is changed to not be the display value.

But I see the point and will make no resistance to it (this would be more relevant especially if strings were to contain spaces)

Reply:
What do you think? I can see this being annoying because of having to convert back and forth

Reply:
Also, what about `TechnologyTypes`? Not considering translation there I guess?

Reply:
`TechnologyTypes` is different, because they don't have translations. In this case we would store the enum values in the schema, and the frontend would fetch a translation document in the db for the selected value, hence the enum-style, so that this is decoupled from the display as much as possible. It is easier to do from the start rather than later.

Reply:
Hmm I see 🤔 

However, I think that fetching the translation document sounds weird - we could just have the translation data in the Front-End using something like [LinguiJS](https://lingui.js.org/) or [react-intl](https://github.com/formatjs/react-intl).
What really matters is that the Front-End adheres to the Schema when making requests, and everything else (display related) should be handled by the Front-End imo

Anyway, this is a bit of a premature discussion anyway, as the time to worry about that is still a long time away haha

Reply:
Translations should not be hardcoded in the front-end, that makes changes harder, as to fix a simple typo in a translation, for example, one would need to create a PR, pass the tests, etc.. versus changing the DB record directly. But yeah, this is another discussion.

Reply:
I don't really agree but yeah, that should be another discussion.

I'm going from what I've seen in more "corporate" environments, in which translation database files were coupled to the front-end, as they are display-related.

Furthermore, something that is displayed relying on a database means that changing it is not as easy, especially if an external database service is used. I believe that keeping this in source-code allows it to be version controlled and thus, more easily acted upon.

With NIAEFEUP/niployments#1 the deploy cycle should become faster and thus this would also be a reduced problem.